src/models/mmt_encoder.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2021 Google LLC
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     https://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+import tensorflow as tf
+from official.modeling import tf_utils
+from official.nlp.keras_nlp import layers
+
+from etcmodel import layers as etc_layers
+from etcmodel import feature_utils as etc_feature_utils
+
+_NUM_OTHER_RELATIVE_IDS = 3
+
+
+class MmtEncoder(tf.keras.Model):
+  """Multimodal Transformer-based Encoder
+
+  This network implements a multimodal transformer-based encoder for 
+  image and language understanding. It includes the embedding lookups and 
+  relative transformer layers, but not the masked language model (mlm),
+  masked patch prediction (mpp), or classification task networks.
+
+  Args: refer to `MmtConfig` for more details.
+    We follow BERT implementation and use approximated gelu function for faster
+    TPU computation.
+    https://github.com/tensorflow/models/blob/2de518be2d6a6e3670b223a4582b1353538d3489/official/nlp/keras_nlp/encoders/bert_encoder.py#L26inner_activation
+    https://github.com/google/jax/issues/4428 
+  """
+
+  def __init__(self,
+               vocab_size: int,
+               segment_vocab_size: int = 16,
+               embedding_size: int = None,
+               hidden_size: int = 768,
+               num_hidden_layers: int = 12,
+               num_attention_heads: int = 12,
+               intermediate_size: int = 3072,
+               inner_activation=lambda x: tf.keras.activations.gelu(
+                   x, approximate=True),
+               hidden_dropout_prob: float = 0.1, 
+               attention_probs_dropout_prob: float = 0.1,
+               max_absolute_position_embeddings: Optional[int] = None,
+               relative_vocab_size: int = 32,
+               relative_pos_max_distance: int = 12,
+               initializer_range: float = 0.02,
+               use_pre_activation_order: bool = False,
+               use_one_hot_lookup: bool = True,
+               name: str = "mmt_encoder",
```

Comment:
nit: the rest of the file uses single quotes, so, switch to single quotes here too for consistency.